src/download.rs

```diff
@@ -66,16 +69,29 @@ fn download_a_chunk(http_client: &Client,
     if !body.check_partialcontent_status() {
         return Err(Error::Status);
     }
-    let mut bytes_buffer = [0; 2048];
+    let mut bytes_buffer = [0; DOWNLOAD_BUFFER_BYTES];
     let mut sum_bytes = 0;
+
+    let progress_update_interval = Duration::from_millis(500);
```

Comment:
Maybe you can extract the 500 value in a constant (like REFRESH_DURATION) - it would be nice if we need to update this one later.